### DISCUSSION TITLE
Alerting mechanism of apache superset does not work at all

### DISCUSSION THREAD

**Discussion by alexkyurtev** [apache/superset#28126] [Q&A / Help]
### Bug description

I am trying to run via docker desktop locally the apache superset alerting mechanism . 
Alerting options does appear , I am also able to create alerts , but unfortunatelly no slack notifications , nor emails are received . 
I can share here the superset_config.py file I`ve done , in order for you to reproduce or troubleshoot together with me . 

### How to reproduce the bug

1. Attached is my superset_config.py config here in order to be easily reproduced . 
2. I am using docker desktop , and successfully run the superset on my laptop.
3. alerting mechanism also appears but no alerts are received through slack or email 
4. Attached is screenshots pointing that no log of the alerts are generated , nothing appears .

### Screenshots/recordings

<img width="1636" alt="image" src="https://github.com/apache/superset/assets/105293655/6874aa20-8d87-47b9-8e5d-8293ba4a8b14">
<img width="1602" alt="image" src="https://github.com/apache/superset/assets/105293655/90830059-3d15-459e-ac0d-a44973db7fe8">

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Firefox

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
The feature definitely works, I'm confident that it's a config/setup issue. Converting this to a Q&A discussion, but I also encourage you to ask on `#deploying-superset` on [slack](http://bit.ly/join-superset-slack)

**Comment by sfirke** [comment]
You say you are running with Docker Desktop.  That sounds like a single container for the application, is that correct?  Superset needs a Celery instance so that a Celery worker can come visit the report, take a screenshot, and send an email.  It also needs a Celery unit to manage the schedule.

In the Docker Compose or Kubernetes deployments, these would be containers.  If you are running a single Docker container for your application, or deploying via PyPI (not your case, but for anyone else reading this), you would need to set up Celery separately.  There is no Superset documentation for this.

I would advise trying out Docker Compose for your deployment, then the Celery setup and connection is (mostly) baked in.  And I think we need to add to the docs to make it clearer up top: Superset uses Celery for this component and if you don't use Docker Compose or k8s, you'll need to handle this on your own.  Which seems daunting to me but I have seen several people in Slack accomplish it.

**Reply by alexkyurtev** [reply]
Thanks for the input . i`ve attached in the above comment my superset_config.py 
The reason we are trying to run it in docker desktop is that we cannot use docker compose to run superset in cloud foundry . Thats why we are trying to stick with docker desktop . So any suggestions for running it with desktop are most welcome.

Regarding the docker compose setup , if i look at the alerting documentation about it : 

You must have in your docker-compose.yml
- A Redis message broker --- > **i think i have this one in my config , right ?** 
- PostgreSQL DB instead of SQLlite  --- > **in my superset , I am reading a SQLlite db , or this is something different ?** 
- One or more celery worker --- > **how can i configure this ?**
- A single celery beat ---- > **how can I configure this ?** 

**In the context of my superset_config , what I need to add exactly to my config?** 


FYI : running these command in my docker desktop superset container :
**_celery --app=superset.tasks.celery_app:app worker --pool=prefork -O fair -c 4
celery --app=superset.tasks.celery_app:app beat_**

And linking my superset container to my redis container with : 
_**docker run p 3100:8088  -name supeset_test --link my-redis-container:my-redis-container -d 8f82eeed6745**_

I managed to get a little progress , now in the UI the alerting : 
<img width="926" alt="image" src="https://github.com/apache/superset/assets/105293655/022590d6-3357-4d27-bbff-d6268678de33">
I get alert running , but still no slack messages , nor emails .

**Reply by sfirke** [reply]
In short you would need to run Redis and Celery in different containers (or on the host machine I guess) and then get Superset connected to them.  Sounds like you are running a Redis container, great.  You can inspect the keys and see if anything is getting stored in there.  As long as you are comfortable with it, you can keep using SQLite as a metadata backend, though most people recommend using Postgres or MySQL if this is an important production deployment.

I don't know how to configure Celery as a standalone service, sorry.  I just use what comes baked into the docker compose deployment.  People say you can set up Flower as a monitoring service for Celery.

**Comment by rusackas** [comment]
You should probably rotate your Slack API token.

**Reply by alexkyurtev** [reply]
the token is active , im sure of it . also if the slack doesn`t work , why does the email doesn`t as well . 
Nothing appears in the execution logs , so again im missing something important : 

<img width="1675" alt="image" src="https://github.com/apache/superset/assets/105293655/c82108c0-db27-41f5-bdd5-559ff1e201b1">

**Reply by rusackas** [reply]
No, I mean you posted the token on this thread, so anyone on the internet can see/use it.
